src/lib/UI/Module/ContentTree/NodeFactory.php

```diff
@@ -179,6 +173,56 @@ private function countSubitems(Location $parentLocation): int
         return $this->searchService->findLocations($searchQuery)->totalCount;
     }
 
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Location[] $containerLocations
+     */
+    private function countAggregatedSubitems(array $containerLocations): array
+    {
+        if (empty($containerLocations)) {
+            return [];
+        }
+
+        if (\count($containerLocations) > self::MAX_AGGREGATED_LOCATION_IDS) {
+            $containerLocationsChunks = array_chunk($containerLocations, self::MAX_AGGREGATED_LOCATION_IDS);
+
+            $result = [];
+            foreach ($containerLocationsChunks as $containerLocationsChunk) {
+                $result = array_replace($result, $this->countAggregatedSubitems($containerLocationsChunk));
+            }
+
+            return $result;
+        }
+
+        $parentLocationIds = array_column($containerLocations, 'id');
+
+        $searchQuery = new LocationQuery();
+        $searchQuery->filter = new Criterion\ParentLocationId($parentLocationIds);
+        $locationChildrenTermAggregation = new Query\Aggregation\Location\LocationChildrenTermAggregation('childrens');
+        $locationChildrenTermAggregation->setLimit(\count($parentLocationIds));
+        $searchQuery->aggregations[] = $locationChildrenTermAggregation;
+
+        $result = $this->searchService->findLocations($searchQuery);
+
+        try {
+            return $this->aggregationResultToArray($result->aggregations->get('childrens'));
+        } catch (OutOfBoundsException $e) {
+        }
```

Comment:
Could you please replace `try ... catch` with `\eZ\Publish\API\Repository\Values\Content\Search\AggregationResultCollection::has(...)` ?